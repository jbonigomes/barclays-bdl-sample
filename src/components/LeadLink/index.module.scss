@import '../../index.scss';

.lead {
  padding: 0;
  border: none;
  outline: none;
  cursor: pointer;
  font-size: 20px;
  background: none;
  text-align: left;
  color: $dark-blue;
  line-height: 28px;
  margin-bottom: 8px;
  font-family: 'ExpertSans-Light';

  &:focus,
  &:hover,
  &:active {
    text-decoration: underline;
  }

  &.lightHover {
    &:focus,
    &:hover,
    &:active {
      color: $blue;
    }
  }

  @media (min-width: $breakpoint) {
    font-size: 24px;
    line-height: 36px;
  }
}
