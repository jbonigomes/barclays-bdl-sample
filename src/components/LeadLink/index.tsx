import React from 'react'
import styles from './index.module.scss'

type LeadLinkProps = {
  text: string,
  lightHover?: boolean,
  onClick(e: React.MouseEvent<HTMLElement>): void,
}

const LeadLink: React.SFC<LeadLinkProps> = ({
  text,
  onClick,
  lightHover,
}: LeadLinkProps) => (
  <div>
    <button
      onClick={onClick}
      className={`${styles.lead} ${lightHover ? styles.lightHover : ''}`}
    >
      {text}
    </button>
  </div>
)

export default LeadLink
