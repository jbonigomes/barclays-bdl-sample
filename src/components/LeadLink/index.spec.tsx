import React from 'react'
import renderer from 'react-test-renderer'

import LeadLink from '.'

test('Match snapshot', () => {
  const props = {
    text: 'text',
    onClick: jest.fn()
  }

  const tree = renderer.create(
    <LeadLink text={props.text} onClick={props.onClick} />
  )

  expect(tree.toJSON()).toMatchSnapshot()
})

test('Match snapshot with light colour on hover', () => {
  const props = {
    text: 'text',
    lightHover: true,
    onClick: jest.fn(),
  }

  const tree = renderer.create(
    <LeadLink
      text={props.text}
      onClick={props.onClick}
      lightHover={props.lightHover}
    />
  )

  expect(tree.toJSON()).toMatchSnapshot()
})
