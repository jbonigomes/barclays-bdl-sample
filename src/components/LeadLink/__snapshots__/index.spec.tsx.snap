// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshot 1`] = `
<div>
  <button
    className="lead "
    onClick={[MockFunction]}
  >
    text
  </button>
</div>
`;

exports[`Match snapshot with light colour on hover 1`] = `
<div>
  <button
    className="lead lightHover"
    onClick={[MockFunction]}
  >
    text
  </button>
</div>
`;
